src/main/java/com/leaf/billing/utils/PropertyBuilder.java

```diff
@@ -15,18 +21,70 @@ private PropertyBuilder() {
     public static final String TARGET_BASE_PATH = "targetBasePath";
     public static final String SOURCE_DELETE_FILES = "sourceDeleteFiles";
 
-    public static JobProperties build(String... args) {
+    public static JobProperties build(String... args) throws IOException, URISyntaxException {
         Map<String, String> sparkProperties = Arrays.stream(args)
                 .filter(arg -> arg.contains("="))
                 .map(arg -> arg.split("=", 2))
                 .filter(keyValue -> keyValue.length == 2)
                 .collect(Collectors.toMap(keyValue -> keyValue[0], keyValue -> keyValue[1]));
+
+        if(!nonePathIsNull(sparkProperties.get(SOURCE_BASE_PATH), sparkProperties.get(TARGET_BASE_PATH))){
+            System.exit(1);
+        }
+        String normalizedPathSource = normalizePath(sparkProperties.get(SOURCE_BASE_PATH));
+        String normalizedPathTarget = normalizePath(sparkProperties.get(TARGET_BASE_PATH));
+        validateEntriesPath(normalizedPathSource, normalizedPathTarget);
         JobProperties jobProperties = new JobProperties();
-        jobProperties.setSourceBasePath(sparkProperties.get(SOURCE_BASE_PATH));
-        jobProperties.setTargetBasePath(sparkProperties.get(TARGET_BASE_PATH));
+        jobProperties.setSourceBasePath(normalizedPathSource);
+        jobProperties.setTargetBasePath(normalizedPathTarget);
         jobProperties.setSourceDeleteFiles(Boolean.valueOf(sparkProperties.getOrDefault(SOURCE_DELETE_FILES, "False")));
         return jobProperties;
     }
 
+    public static boolean nonePathIsNull(String source, String target){
+        return !(source == null || target == null);
+    }
```

Comment:
```suggestion
    public static boolean PathIsNotNull(String path){
        return !(path == null);
    }
```
something like this maybe? but it is so small I don't know if is worth to keep haha

Reply:
Thanks, Ricardo! This method doesn't exist more because I implemented your first suggestion on this commit: https://github.com/Leaf-Agriculture/billing-data-aggregation-job/pull/6/commits/c35585ec866ae475dac2134ad2c793d4c2ae0261